packages/fxa-payments-server/src/routes/Product/index.test.tsx

```diff
@@ -0,0 +1,514 @@
+import React from 'react';
+import { render, cleanup, act, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import nock from 'nock';
+
+import { PAYMENT_ERROR_2 } from '../../lib/errors';
+import {
+  defaultAppContextValue,
+  MockApp,
+  MockStripeType,
+  setupMockConfig,
+  mockConfig,
+  mockServerUrl,
+  mockStripeElementOnChangeFns,
+  elementChangeResponse,
+} from '../../lib/test-utils';
+
+import { SignInLayout } from '../../components/AppLayout';
+import Product from './index';
+
+describe('routes/Product', () => {
+  let authServer = '';
+  let profileServer = '';
+
+  beforeEach(() => {
+    setupMockConfig({
+      ...mockConfig,
+      productRedirectURLs: PRODUCT_REDIRECT_URLS,
+    });
+    authServer = mockServerUrl('auth');
+    mockOptionsResponses(authServer);
+    profileServer = mockServerUrl('profile');
+    mockOptionsResponses(profileServer);
+  });
+
+  afterEach(() => {
+    nock.cleanAll();
+    return cleanup();
+  });
+
+  const Subject = ({
```

Comment:
This is kind of the monster of the whole PR. I just kept writing tests to cover cases in the payment page until test coverage hit 100%, so it's a bit exhaustive and hits all the permutations of the UI. I tried to minimize the copy/paste in common shared functions, but it's still a lot of code 😓